kratos/sources/constitutive_law.cpp

```diff
@@ -635,25 +635,29 @@ void ConstitutiveLaw::CalculateMaterialResponseCauchy (Parameters& rValues)
 
     void ConstitutiveLaw::InitializeMaterialResponse(Parameters& rValues,const StressMeasure& rStressMeasure)
     {
-      switch(rStressMeasure)
-	{
-	case StressMeasure_PK1:         InitializeMaterialResponsePK1(rValues);
-	  break;
+        if (!RequiresInitializeMaterialResponse()) {

```

Comment:
OK, so your plan is to info to the element developers, not only CL developers...I see

Reply:
i disagree... this is a possible optimization, it is not mandatory